grader_host/index.js

```diff
@@ -622,7 +622,7 @@ function uploadResults(info, callback) {
         const params = {
           Bucket: s3Bucket,
           Key: `${s3RootKey}/results.json`,
-          Body: Buffer.from(JSON.stringify(results, null, '  '), 'binary'),
+          Body: Buffer.from(JSON.stringify(results, null, 2)),
```

Comment:
I'm impressed that you tracked down the issue to this line (which is clearly wrong, to be fair).

Reply:
I ultimately compared `results.json` from the job tarball to the raw `results.json` that we upload to S3 and noticed that they had different encodings 🙃